src/basic/uid-range.h

```diff
@@ -8,7 +8,10 @@ typedef struct UidRange {
         uid_t start, nr;
 } UidRange;
 
-int uid_range_add(UidRange **p, size_t *n, uid_t start, uid_t nr);
+int uid_range_add_internal(UidRange **p, size_t *n, uid_t start, uid_t nr, bool coalsece);
+static inline int uid_range_add(UidRange **p, size_t *n, uid_t start, uid_t nr) {
+        return uid_range_add_internal(p, n, start, nr, true);
+}
```

Comment:
hmm, why not make uid_range_add_internal() a static function, and drop the coalesce parameter? And then make uid_range_add() a regular public function that calls uid_range-add_internal() and then itself calls uid_range_coalesce()?

Reply:
That makes sense. but in the revised version several tests for `uid_range_coalesce()` are added. See the last commit.